tests/sentry/replays/test_project_replay_details.py

```diff
@@ -1,19 +1,30 @@
 import datetime
 from io import BytesIO
+from unittest import mock
 from uuid import uuid4
 
+import pytest
 from django.urls import reverse
 
 from sentry.models import File
+from sentry.replays import tasks
 from sentry.replays.models import ReplayRecordingSegment
 from sentry.replays.testutils import assert_expected_response, mock_expected_response, mock_replay
 from sentry.testutils import APITestCase, ReplaysSnubaTestCase
 from sentry.testutils.helpers import TaskRunner
 from sentry.testutils.silo import region_silo_test
+from sentry.utils import kafka_config
 
 REPLAYS_FEATURES = {"organizations:session-replay": True}
 
 
+@pytest.fixture(autouse=True)
+def setup():
```

Comment:
i'm kind of confused as to what this does? this test doesn't use kafka?

Reply:
This tests does end up touching kafka behind the scenes and the `flush()` call [here][flush-call] causes a 5min delay in the test. But I agree that it doesn't check for any use of kafka or apply any assertions.

So how does the `test_delete` test result in the `flush()` call, you can see it from the profile, which I've now attached to the PR description, but it roughly boils down to:

1. [test_delete][test-del-fn] calls the client delete
1. The request ends up in [project_replay_details][proj-replay-call]
1. This results in [delete_recording_segments][del-seg-call] getting called
1. That calls [_archive_replay][archive-replay-call]
1. Which creates a [kafka publisher][publish-call] and calls `publish()` which calls [flush][flush-call]

[flush-call]: https://github.com/getsentry/sentry/blob/master/src/sentry/utils/pubsub.py#L77
[publish-call]: https://github.com/getsentry/sentry/blob/master/src/sentry/replays/tasks.py#L51-L52
[archive-replay-call]: https://github.com/getsentry/sentry/blob/master/src/sentry/replays/tasks.py#L36-L65
[del-seg-call]: https://github.com/getsentry/sentry/blob/master/src/sentry/replays/tasks.py#L21-L24
[proj-replay-call]: https://github.com/getsentry/sentry/blob/master/src/sentry/replays/endpoints/project_replay_details.py#L60-L73
[test-del-fn]: https://github.com/getsentry/sentry/blob/master/tests/sentry/replays/test_project_replay_details.py#L161

Reply:
got it. thx.